Improved performance for allocators

agibsonccc opened the pull request:
## What changes were proposed in this pull request?
Fix inplace operations setting output arrays
Fix misc typos in error messages
Clean up op context commented code
Add new mass set of input/output args for contexts allowing setting to happen all at once in c++ instead of once at a time in java
Get rid of createShapeInfo duplicate allocation. Change to inplace model for generating shape buffers.
Fix up cbow to be parity with skipgram
(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

## Quick checklist

The following checklist helps ensure your PR is complete:

- [ X] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [X ] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [X ] Created tests for any significant new code additions.
- [ X] Relevant tests for your changes are passing.

agibsonccc committed:
Fix lombok Thread.builder clash with jdk 17

agibsonccc committed:
Fix inplace operations setting output arrays
Fix misc typos in error messages
Clean up op context commented code
Add new mass set of input/output args for contexts allowing setting to happen all at once in c++ instead of once at a time in java

agibsonccc committed:
Fix inplace operations setting output arrays
Fix misc typos in error messages
Clean up op context commented code
Add new mass set of input/output args for contexts allowing setting to happen all at once in c++ instead of once at a time in java
Get rid of createShapeInfo duplicate allocation. Change to inplace model for generating shape buffers.
Fix up

agibsonccc committed:
Fix inplace operations setting output arrays
Fix misc typos in error messages
Clean up op context commented code
Add new mass set of input/output args for contexts allowing setting to happen all at once in c++ instead of once at a time in java
Get rid of createShapeInfo duplicate allocation. Change to inplace model for generating shape buffers.
Fix up cbow to be parity with skipgram

agibsonccc committed:
Fix inplace operations setting output arrays
Fix misc typos in error messages
Clean up op context commented code
Add new mass set of input/output args for contexts allowing setting to happen all at once in c++ instead of once at a time in java
Get rid of createShapeInfo duplicate allocation. Change to inplace model for generating shape buffers.
Fix up cbow to be parity with skipgram
Fix training iterations
Convert cbow constructor usage to builder

treo commented on deeplearning4j/deeplearning4j-nlp-parent/deeplearning4j-nlp/src/main/java/org/deeplearning4j/models/embeddings/learning/impl/elements/CBOW.java:
```diff
@@ -178,133 +178,223 @@ public boolean isEarlyTerminationHit() {
+                        .numWorkers(workers)
+                        .iterations(useInference ? configuration.getIterations() * configuration.getEpochs() : 1)
+                        .build();
+            }
```
All of those builder repetitions look fishy. Why not create the builder with the common options set and then set the options that are actually different?

agibsonccc committed:
Fix inplace operations setting output arrays
Fix misc typos in error messages
Clean up op context commented code
Add new mass set of input/output args for contexts allowing setting to happen all at once in c++ instead of once at a time in java
Get rid of createShapeInfo duplicate allocation. Change to inplace model for generating shape buffers.
Fix up cbow to be parity with skipgram
Fix training iterations
Convert cbow constructor usage to builder
Consolidate CBOW builder
